### PR TITLE
Fixing the attention mask in test, test_qwen2_7b_sdpa_input_alignment…

### LINKED ISSUE #163689: DISABLED test_qwen2_7b_sdpa_input_alignment_requires_recompile (__main__.CudaReproTests)
Platforms: rocm

This test was disabled because it is failing on main branch ([recent examples](https://torch-ci.com/failure?failureCaptures=%5B%22inductor%2Ftest_cuda_repro.py%3A%3ACudaReproTests%3A%3Atest_qwen2_7b_sdpa_input_alignment_requires_recompile%22%5D)).

Caused by https://github.com/pytorch/pytorch/pull/163083.

cc @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @jataylo @hongxiayang @naromero77amd @pragupta @jerrymannil @xinyazhang @mruberry @dllehr-amd

### PR DESCRIPTION
**Pull request by umechand-amd**
Fixes #163689
This is a PR to fix the test, test_qwen2_7b_sdpa_input_alignment_requires_recompile>
All the values in the mask tensor were False which led to the bias being -inf which led to NaNs in the final output.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

**Comment by umechand-amd** [comment]
Actually, this test still failed due to NaNs in output. Regardless, the mask should never be all 0s (False)so we should merge this PR while I continue debugging the test.
